src/Nancy/HttpLinkRelation.cs

```diff
@@ -0,0 +1,180 @@
+﻿namespace Nancy
+{
+    using System;
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// The relation that identifies the semantics of a <see cref="HttpLink"/> contained in an RFC 5899 Link HTTP header,
+    /// as built by the <see cref="HttpLinkBuilder"/>.
+    /// </summary>
+    /// <seealso cref="System.Uri" />
+    [Serializable]
+    public class HttpLinkRelation : Uri, IEquatable<HttpLinkRelation>
```

Comment:
It's a conditionally relative or absolute `Uri`. That's what the code in `Parse()` deals with. According to RFC 5899, all `rel` values not [registered in IANA](http://www.iana.org/assignments/link-relations/link-relations.xhtml), should be prefixed with a base URI, while `rel` values registered in IANA should be unprefixed, opaque strings.